src/AstroImages.jl

```diff
@@ -47,31 +49,43 @@ for n in (8, 16, 32, 64)
     end
 end
 
-struct AstroImage{T<:Real,C<:Color}
-    data::Matrix{T}
+struct AstroImage{T<:Real,C<:Color, N}
+    data::NTuple{N, Matrix{T}}
 end
 
 """
     AstroImage([color=Gray,] data::Matrix{Real})
+    AstroImage(color::Type{<:Color}, data::NTuple{N, Matrix{T}}) where {T<:Real, N}
 
 Construct an `AstroImage` object of `data`, using `color` as color map, `Gray` by default.
 """
 AstroImage(color::Type{<:Color}, data::Matrix{T}) where {T<:Real} =
-    AstroImage{T,color}(data)
-AstroImage(data::Matrix{T}) where {T<:Real} = AstroImage{T,Gray}(data)
+    AstroImage{T,color, 1}(ntuple(i-> data, 1))
+AstroImage(color::Type{<:Color}, data::NTuple{N, Matrix{T}}) where {T<:Real, N} =
+    AstroImage{T,color, N}(data)
+AstroImage(data::Matrix{T}) where {T<:Real} = AstroImage{T,Gray,1}(ntuple(i-> data, 1))
+AstroImage(data::NTuple{N, Matrix{T}}) where {T<:Real, N} = AstroImage{T,Gray,N}(data)
 
 """
     AstroImage([color=Gray,] filename::String, n::Int=1)
+    AstroImage(color::Type{<:Color}, file::String, n::NTuple{N, Int}) where {N}
 
 Create an `AstroImage` object by reading the `n`-th extension from FITS file `filename`.
+
 Use `color` as color map, this is `Gray` by default.
 """
 AstroImage(color::Type{<:Color}, file::String, ext::Int) =
     AstroImage(color, load(file, ext))
+AstroImage(color::Type{<:Color}, file::String, ext::NTuple{N, Int}) where {N} =
+    AstroImage(color, load(file, ext))
+
 AstroImage(file::String, ext::Int) = AstroImage(Gray, file, ext)
+AstroImage(file::String, ext::NTuple{N, Int}) where {N} = AstroImage(Gray, file, ext)
```

Comment:
Why forcing `Gray`?  `Gray` it's suitable for `N = 1`, for `N = 3` I'm not sure it'd be good

Reply:
This PR is not supposed to enable support for RGB yet. Therefore  `Color` hasn't been utilized anywhere. As for forcing `Gray`, all the images are in the form of 2D matrix, hence `Gray` (`RGB` or others are for higher dimensions). 
When I add support for RGB, then I will make amends to handle `Color`. This PR just enables the ability to hold more than one `ImageHDU`.